src/pyhf/tensor/pytorch_backend.py

```diff
@@ -121,13 +132,7 @@ def astensor(self, tensor_in, dtype='float'):
             log.error('Invalid dtype: dtype must be float, int, or bool.')
             raise
 
-        tensor = torch.as_tensor(tensor_in, dtype=dtype)
-        # Ensure non-empty tensor shape for consistency
-        try:
-            tensor.shape[0]
-        except IndexError:
-            tensor = tensor.expand(1)
-        return tensor
+        return torch.as_tensor(tensor_in, dtype=dtype)
```

Comment:
This is going to require a solution for avoiding `TypeError: iteration over a 0-d tensor` for empty-tensors.